frontend/src/components/OdhAppCard.tsx

```diff
@@ -105,7 +105,11 @@ const OdhAppCard: React.FC<OdhAppCardProps> = ({ odhApp }) => {
     <CardFooter className="odh-card__footer">
       {odhApp.metadata.name === 'jupyter' ? (
         odhApp.spec.internalRoute ? (
-          <Link to={odhApp.spec.internalRoute} className="odh-card__footer__link">
+          <Link
+            data-testid="jupyter-app-link"
+            to={`/${odhApp.spec.internalRoute}`}
+            className="odh-card__footer__link"
+          >
             Launch application
           </Link>
         ) : null
```

Comment:
Weird legacy code... Lets just change this to `/notebookController` please (instead of using the spec property).

We have 5 other instances that hardcode the path. JH was a wild alternative back in the day and it needed more dynamic nature. This is an internal route, we should just sync it up imo.